packages/aws-cdk-lib/aws-appsync/lib/auth-config.ts

```diff
@@ -0,0 +1,214 @@
+   * A regular expression can be specified so AppSync can validate against multiple client identifiers at a time.
+   * @example - 'ABCD|CDEF' // where ABCD and CDEF are two different clientId
+   * @default - * (All)
+   */
+  readonly clientId?: string;
+  /**
+   * The issuer for the OIDC configuration. The issuer returned by discovery must exactly match the value of `iss` in the OIDC token.
+   */
+  readonly oidcProvider: string;
+}
+
+/**
+ * Configuration for Lambda authorization in AppSync. Note that you can only have a single AWS Lambda function configured to authorize your API.
+ */
+export interface AppSyncLambdaAuthorizerConfig {
+  /**
+   * The authorizer lambda function.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-graphqlapi-lambdaauthorizerconfig.html
+   */
+  readonly handler: IFunction;
+
+  /**
+   * How long the results are cached.
+   * Disable caching by setting this to 0.
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly resultsCacheTtl?: Duration;
+
+  /**
+   * A regular expression for validation of tokens before the Lambda function is called.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
+}
+
+/**
+ * Set up OIDC Authorization configuration for GraphQL APIs and Event APIs
+ */
+export function setupOpenIdConnectConfig(config?: AppSyncOpenIdConnectConfig) {
+  if (!config) return undefined;
+  return {
+    authTtl: config.tokenExpiryFromAuth,
+    clientId: config.clientId,
+    iatTtl: config.tokenExpiryFromIssue,
+    issuer: config.oidcProvider,
+  };
+}
+
+/**
+ * Set up Cognito Authorization configuration for Event APIs
+ */
+export function setupCognitoConfig(config?: AppSyncCognitoConfig) {
+  if (!config) return undefined;
+  return {
+    userPoolId: config.userPool.userPoolId,
+    awsRegion: config.userPool.env.region,
+    appIdClientRegex: config.appIdClientRegex,
+  };
+}
+
+/**
+ * Set up Lambda Authorization configuration for GraphQL APIs and Event APIs
+ */
+export function setupLambdaAuthorizerConfig(config?: AppSyncLambdaAuthorizerConfig) {
+  if (!config) return undefined;
+  return {
+    authorizerResultTtlInSeconds: config.resultsCacheTtl?.toSeconds(),
+    authorizerUri: config.handler.functionArn,
+    identityValidationExpression: config.validationRegex,
+  };
+}
+
+/**
+ * Create an API Key for GraphQL APIs and Event APIs
+ */
+export function createAPIKey(scope: Construct, apiId: string, config?: AppSyncApiKeyConfig) {
+  if (config?.expires?.isBefore(Duration.days(1)) || config?.expires?.isAfter(Duration.days(365))) {
+    throw Error('API key expiration must be between 1 and 365 days.');
+  }
+  const expires = config?.expires ? config?.expires.toEpoch() : undefined;
+  return new CfnApiKey(scope, `${config?.name || 'Default'}ApiKey`, {
+    expires,
+    description: config?.description,
+    apiId: apiId,
+  });
+}
```

Comment:
This makes sense. Right now the only difference between the GraphQL and Event API is the`UserPoolConfig` and `AppSyncCognitoConfig` resources. The`UserPoolConfig` has that additional parameter for `defaultAction`. This is reflected in the CloudFormation as well - both having different types.

I do like the idea of the interface for future extensibility. I'll implement.

Reply:
Let me know what you think of the implementation on this. It seems a bit redundant implemented in the `eventapi.ts` file and the naming is a bit odd since I already have the `EventApiAuthConfig` interface.

Reply:
Thanks @kwwendt for the implementation. Overall looks good. Just left a minor comment regarding the return type for the methods in the interface.